metomi/rose/macros/rule.py

```diff
@@ -455,11 +463,17 @@ def _process_rule(
                 x_id_num_str = search_result.replace("this", "").strip('()')
                 key = self.INTERNAL_ID_THIS_SETTING.format(x_id_num_str)
                 local_map[key] = value_string
+            search_result = search_result.replace(
+                'template variables', 'template_variables')
             rule = rule.replace(search_result, key, 1)
 
         # Replace ids (namelist:foo=bar) with their cast values.
```

Comment:
You've added this `search_result.replace` fix into most, but not all of the sections in this method.

These two remain unpatched:

```
# Number-like-strings into numbers.
# Strings into proper string variables.
```

It's ok if they are not needed there, but could you explain why it's not needed there for review.

Reply:
Those loops where I've made a change the Regex will return `template variables=something` - the loops where I haven't the regexes only return the bit after the `=`:

```python
# For example the following are broken with `template variables`
>>> RuleEvaluator.REC_LEN_FUNC.findall('{% if len(template variables=FOO) == 1 %}True{% else %}False{% endif %}')
[]
>>> RuleEvaluator.REC_CONFIG_ID.findall('{% if len(template variables=FOO) == 1 %}True{% else %}False{% endif %}')
['variables=FOO']

# And fixed by adding `_`
>>> RuleEvaluator.REC_CONFIG_ID.findall('{% if template_variables=FOO == "2e1" %}True{% else %}False{% endif %}')
['template_variables=FOO']
>>> RuleEvaluator.REC_LEN_FUNC.findall('{% if len(template_variables=FOO) == 1 %}True{% else %}False{% endif %}')
[(' ', 'template_variables=FOO', ' ')]

# Wheras
>>> RuleEvaluator.REC_SCI_NUM.findall('{% if template_variables=FOO == "2e1" %}True{% else %}False{% endif %}')
['2e1']
>>> RuleEvaluator.REC_VALUE.findall('{% if template_variables=FOO == "2e1" %}True{% else %}False{% endif %}')
['"2e1"']

Reply:
Makes sense, just needed to understand why.